crates/core/src/models/signature.rs

```diff
@@ -0,0 +1,33 @@
+use reth_rpc_types::Signature;
+use starknet::core::types::FieldElement;
+use thiserror::Error;
+
+use super::felt::Felt252Wrapper;
+
+#[derive(Debug, Error)]
+pub enum StarknetSignatureError {
+    #[error("missing Starknet signature param {0}")]
+    MissingSignatureParamsError(String),
+}
+
+pub struct StarknetSignature(Vec<FieldElement>);
+
+impl From<Vec<FieldElement>> for StarknetSignature {
```

Comment:
are we doing this because Starknet signatures are non-opinionated on the size/ format? it's simply a vec of felts

Reply:
Yes indeed, I think that's needed for account abstraction (example for [InvokeTransaction](https://github.com/xJonathanLEI/starknet-rs/blob/master/starknet-core/src/types/codegen.rs#L592)).